lib/posix/posix_internal.h

```diff
@@ -15,35 +15,25 @@
  */
 #define PTHREAD_OBJ_MASK_INIT 0x80000000
 
-enum pthread_state {
-	/* The thread is running and detached. */
-	PTHREAD_DETACHED = PTHREAD_CREATE_DETACHED,
-	/* The thread is running and joinable. */
-	PTHREAD_JOINABLE = PTHREAD_CREATE_JOINABLE,
-	/* The thread structure is unallocated and available for reuse. */
-	PTHREAD_TERMINATED,
-	/* A joinable thread exited and its return code is available. */
-	PTHREAD_EXITED
-};
-
 struct posix_thread {
 	struct k_thread thread;
 
+	/* List node for ready_q, run_q, or done_q */
+	sys_dnode_t q_node;
+
 	/* List of keys that thread has called pthread_setspecific() on */
 	sys_slist_t key_list;
 
 	/* Exit status */
 	void *retval;
 
-	/* Pthread cancellation */
-	int cancel_state;
-	int cancel_pending;
-	struct k_spinlock cancel_lock;
+	struct k_spinlock lock;
 
-	/* Pthread State */
-	enum pthread_state state;
-	pthread_mutex_t state_lock;
-	pthread_cond_t state_cond;
+	/* Flags */
```

Comment:
There are four bitfields here that are all <= 8 bits. I'd use bytes instead of bits as that's a lot faster and will consume no more memory.